### PR TITLE
Update nsxt_nat_rule with firewall_match

### PR DESCRIPTION
**Pull request by ksamoray**
firewall_match attribute replace nat_pass in nsxt_nat_rule resource.

Fixes: #945

### CODE REVIEW

File: nsxt/resource_nsxt_nat_rule.go [hunk 1]
```diff
@@ -245,41 +239,43 @@ func resourceNsxtNatRuleUpdate(d *schema.ResourceData, m interface{}) error {
 	//match_service := d.Get("match_service").(*NsServiceElement)
 	matchSourceNetwork := d.Get("match_source_network").(string)
 	natPass := d.Get("nat_pass").(bool)
+	firewallMatch := model.NatRule_FIREWALL_MATCH_MATCH_INTERNAL_ADDRESS
```
Comment (annakhm) [inline nsxt/resource_nsxt_nat_rule.go:242]: do we know what version this was added in?

**Review by annakhm — COMMENTED** [review]
LGTM

File: nsxt/resource_nsxt_nat_rule.go (hunk 1 above)
Comment (ksamoray) [reply nsxt/resource_nsxt_nat_rule.go:242]: nat_pass was removed at 4.0.0
firewall_match exists in 2.4 - it seemed old enough so I didn't check earlier versions...

Comment (ksamoray) [reply nsxt/resource_nsxt_nat_rule.go:242]: Maybe I should keep the code which uses the old API for pre-4.0.0 environments?

Comment (annakhm) [reply nsxt/resource_nsxt_nat_rule.go:242]: I think current code should be good, I would run the test against 3.0.0 to be sure

**Review by annakhm — APPROVED** [review]

**Pull request merged**